File: 21551-naveen-siva-kumar/week7/1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeInBetween(ListNode* list1, int a, int b, ListNode* list2) {
        
        ListNode* fast = list1;
        ListNode* slow = list1;
        ListNode* temp = list2;
        
        int count=0;
        while(temp->next!=NULL){
            temp=temp->next;
        }
        
        ListNode* sentinal = new ListNode(0,list1);
        
        while(count!=a-1){
            count++;
            fast=fast->next;
        }
        
        slow = fast;
        fast=fast->next;count++;
        slow->next=list2;
        while(count!=b){ fast=fast->next;count++;}
        temp->next=fast->next;
        
        return sentinal->next;
    }
};

File: 21557 - Mukesh J/Week-7/1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        ListNode dummy= new ListNode(0),slow=dummy,p=null,q=null;
        slow.next=list1;
        int count=-1;
        while(slow.next!=null)
        {
            if(count==a-1)
            {
                p= slow.next;
                slow.next=list2;
                slow=slow.next;
                while(slow.next!=null)
                    slow=slow.next;
                q=slow;
                slow=p;
                count++;
            }
            if(count==b)
            {
                q.next=slow.next;
                break;
            }
            count++;
            slow=slow.next;
        }
        return list1;
    }
}